### PR TITLE
get metadata from REST channel

### PR DESCRIPTION
**Pull request by rgstephens**
**Proposed changes**:
@ancalita The REST channel should by default load the `metadata` key.  I thought this worked in the past but can't seem to find when it was dropped.

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Review by ancalita — CHANGES_REQUESTED** [review]
Good spot, I also couldn't find anything in closed PRs, docs or git blame, maybe it was never implemented 🤔 
Could you please add a changelog entry and relevant test?

File: docs/docs/connectors/your-own-website.mdx [hunk 1]
```diff
@@ -82,7 +82,10 @@ After making the `callback` input available, you can `POST` messages to
 ```json
 {
   "sender": "test_user",  // sender ID of the user sending the message
-  "message": "Hi there!"
+  "message": "Hi there!",
+  "message": {            // optional metadata
```
Comment (ancalita) [inline docs/docs/connectors/your-own-website.mdx:86]: Shouldn't this key be `metadata` instead of `message`?

Comment (rgstephens) [reply docs/docs/connectors/your-own-website.mdx]: Oops, fixed, thanks.

File: rasa/core/channels/rest.py [hunk 2]
```diff
@@ -58,6 +58,21 @@ def _extract_message(self, req: Request) -> Optional[Text]:
...
+
+        Args:
+            request: incoming request with the message of the user
+
+        Returns:
+            Metadata which was extracted from the request.
+        """
+        return request.json.get("metadata", None)
+
     def stream_response(
```
Comment (ancalita) [inline rasa/core/channels/rest.py:76]: Could you please add the missing docstring here so the code quality checks pass 🙏🏻

**Review by ancalita — APPROVED** [review]

File: rasa/core/channels/rest.py (hunk 2 above)
Comment (rgstephens) [reply rasa/core/channels/rest.py:76]: I don't understand the error after reading up on flake8 and public method docstring requirements. I see lines 62-73 look like a valid docstring on a public method but clearly I'm wrong about that.

Comment (ancalita) [reply rasa/core/channels/rest.py:76]: Oh the linter is complaining about the `stream_response` method lacking a docstring. I know you haven't updated this method, however it just seems the linter will flag these past issues in the module you're working in.

Comment (rgstephens) [reply rasa/core/channels/rest.py:76]: Thanks, fixed

**Review by ancalita — APPROVED** [review]

**Comment by rgstephens** [comment]
@ancalita do you know if this will be merged into the next 3.2.x release?

**Comment by ancalita** [comment]
> @ancalita do you know if this will be merged into the next 3.2.x release?

I see this is targeting `main` actually, I'd recommend you rebase to `3.2.x` and edit target branch on GH for the change to make it in the next micro release.

File: changelog/11311.fix.md [hunk 3]
```diff
@@ -0,0 +1 @@
+Re-instates the REST channel metadata feature. Metadata can be provided on the `metadata` key.
```
Comment (ancalita) [inline changelog/11311.fix.md:1]: The file title should contain `bugfix`

**Comment by rgstephens** [comment]
Closing this. Rebasing on `3.2.x` and opening #11436.

**Pull request closed without merging**